pkg/util/merge/merge.go

```diff
@@ -430,16 +431,21 @@ func VolumeMounts(original, override []corev1.VolumeMount) []corev1.VolumeMount
 	}
 
 	sort.SliceStable(mergedMounts, func(i, j int) bool {
-		return mergedMounts[i].Name < mergedMounts[j].Name
+		return volumeMountToString(mergedMounts[i]) < volumeMountToString(mergedMounts[j])
 	})
 
 	return mergedMounts
 }
 
+// volumeMountToString returns a string consisting of all components of the given VolumeMount.
+func volumeMountToString(v corev1.VolumeMount) string {
+	return strings.Join([]string{v.Name, v.MountPath, v.SubPath}, "-")
```

Comment:
[q] Is it guaranteed that none of `Name`, `MountPath`, and `SubPath` will contain the "-" character? The reason I ask is so we don't get a case where we have two different (valid) volume mounts with the same `volumeMountToString()` value, e.g.:
```
Name: a-b
MountPath: c
SubPath: d
```
and 
```
Name: a
MountPath: b-c
SubPath: d
```
Will both have volumeMountToString: `a-b-c-d`

Reply:
good point! Maybe we can change it to include a character that will not be able to appear in the names/paths